test/user_program_data.s

```diff
@@ -0,0 +1,11 @@
+.section .text
+.globl entry
+entry:
+    mov item1, %eax
+    mov item2, %ebx
+loop:
+    jmp loop
+
+.section .data
+item1: .long 0xCAFE
+item2: .long 0xBEEF
```

Comment:
This doesn't have to be for this PR, can make an issue for future testing
Could we have another program that writes.
Could try to write to a .rodata section and make sure the protections are there.

Reply:
Good idea. I'll make an issue post-merge